### PR TITLE
fix: sync refs correctly with `useSyncRef` hook

### PR DESCRIPTION
**Pull request by Lukas742**
This PR fixes the issue of `useConsolidatedRef` returning a ref which is one cicle "late". It implements the new `useSyncRef` hook, which is now returning the actual current node. `useConsolidatedRef` is therefore now deprecated and will be removed with the next minor release.

### CODE REVIEW

File: packages/base/src/hooks/useSyncRef.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { MutableRefObject, Ref, RefCallback, RefObject, useRef } from 'react';
+
+export function useSyncRef<RefType = any>(ref: Ref<RefType>): [RefCallback<RefType>, RefObject<RefType>] {
+  const localRef = useRef<RefType>();
```
Comment (MarcusNotheis) [inline packages/base/src/hooks/useSyncRef.ts:4]:
```suggestion
  const localRef = useRef<RefType>(null);
```
I guess this should be called like this to return a `MutableRefObject`

File: packages/base/src/hooks/useSyncRef.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { MutableRefObject, Ref, RefCallback, RefObject, useRef } from 'react';
+
+export function useSyncRef<RefType = any>(ref: Ref<RefType>): [RefCallback<RefType>, RefObject<RefType>] {
```
Comment (MarcusNotheis) [inline packages/base/src/hooks/useSyncRef.ts:3]:
```suggestion
export function useSyncRef<RefType = any>(ref: Ref<RefType>): [RefCallback<RefType>, MutableRefObject <RefType>] {
```
isn't this rather a MutableRefObject?

File: packages/main/src/components/AnalyticalTable/index.tsx [hunk 3]
```diff
@@ -539,7 +540,10 @@ const AnalyticalTable = forwardRef((props: AnalyticalTablePropTypes, ref: Ref<HT
     return props.data;
   }, [props.data, minRows]);
 
-  const tableInstanceRef = useConsolidatedRef<Record<string, any>>(tableInstance);
+  let tableInstanceRef = useRef<Record<string, any>>();
+  if (tableInstance) {
+    tableInstanceRef = tableInstance;
+  }
```
Comment (MarcusNotheis) [inline packages/main/src/components/AnalyticalTable/index.tsx:546]: This is technically a breaking change - we have accepted a callback ref for the `tableInstance` before that change

File: packages/main/src/components/FilterBar/FilterBar.test.tsx [hunk 4]
```diff
@@ -500,7 +500,7 @@ describe('FilterBar', () => {
     const dialogGoButton = screen.getAllByText('Go')[1];
     fireEvent.click(dialogGoButton);
     expect(onGo).toHaveBeenCalledTimes(2);
-    expect(onFiltersDialogSave).toHaveBeenCalledTimes(1);
+    expect(onFiltersDialogSave).toHaveBeenCalledTimes(2);
```
Comment (MarcusNotheis) [inline packages/main/src/components/FilterBar/FilterBar.test.tsx:503]: why did those counters increase?

Comment (Lukas742) [reply packages/main/src/components/FilterBar/FilterBar.test.tsx:503]: Before, onGo just closed the dialog and din't update the values inside the FilterBar. Now, when the "Go" button is clicked, it's also saving the values and therefore is also calling the corresponding callback.

File: packages/main/src/components/AnalyticalTable/index.tsx [hunk 5]
```diff
@@ -539,7 +540,7 @@ const AnalyticalTable = forwardRef((props: AnalyticalTablePropTypes, ref: Ref<HT
     return props.data;
   }, [props.data, minRows]);
 
-  const tableInstanceRef = useConsolidatedRef<Record<string, any>>(tableInstance);
+  let tableInstanceRef = useRef<Record<string, any>>();
```
Comment (MarcusNotheis) [inline packages/main/src/components/AnalyticalTable/index.tsx:543]:
```suggestion
  const tableInstanceRef = useRef<Record<string, any>>(null);
```
`tableInstanceRef` is never reassigned, so I would prefer const. In addition, using null as default value makes it a `MutableRefObject` :)

**Pull request merged**